Makefile

```diff
@@ -506,7 +506,7 @@ PIC_OBJS = \
 # to CC.  This is a workaround for a Windows command line size limitation.  See
 # the `%.a` rule below for details.
 $(SYSROOT_LIB)/%.so: $(OBJDIR)/%.so.a $(BUILTINS_LIB)
-	$(CC) -nostdlib -shared -o $@ -Wl,--whole-archive $< -Wl,--no-whole-archive $(BUILTINS_LIB)
+	$(CC) -shared -o $@ -Wl,--whole-archive $< -Wl,--no-whole-archive $(BUILTINS_LIB)
```

Comment:
Do we want `-nolibc` perhaps?  (We wouldn't want the static version of libc linked into the dynamic version.. but I'm not sure if that would actually happen..).

Reply:
(Can you see of `-lc` is added to the link line.. I don't think we would want that when building libc itself).

Reply:
Looks like `-lc` does show up:
```
"/Users/dicej/p/component-linking-demo/wasi-sdk/build/install/opt/wasi-sdk/bin/wasm-ld" -m wasm32 -L/Users/dicej/p/component-linking-demo/wasi-sdk/build/install/opt/wasi-sdk/bin/../share/wasi-sysroot/lib/wasm32-wasi /Users/dicej/p/component-linking-demo/wasi-sdk/build/install/opt/wasi-sdk/bin/../share/wasi-sysroot/lib/wasm32-wasi/crt1-reactor.o --entry _initialize -shared --whole-archive build/wasm32-wasi/libc.so.a --no-whole-archive /Users/dicej/p/component-linking-demo/wasi-sdk/build/install/opt/wasi-sdk/lib/clang/17/lib/wasi/libclang_rt.builtins-wasm32.a -lc /Users/dicej/p/component-linking-demo/wasi-sdk/build/install/opt/wasi-sdk/lib/clang/17/lib/wasi/libclang_rt.builtins-wasm32.a -o /Users/dicej/p/component-linking-demo/wasi-sdk/build/install/opt/wasi-sdk/share/wasi-sysroot/lib/wasm32-wasi/libc.so
```
There's no obvious ill effect, but I can go ahead and add `-nolibc`.

Reply:
Apparently I'm doing it wrong:

Reply:
Oh.. maybe clang doesn't support that flag?

Reply:
Maybe `-nodefaultlibs`

Reply:
Yeah, `-nodefaultlibs` seems to do the trick.  I really wish `clang --help` would list _all_ the options it supports.